pkg/client/unversioned/remotecommand/remotecommand.go

```diff
@@ -21,141 +21,118 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 	"sync"
 
 	"k8s.io/kubernetes/pkg/api"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
-	"k8s.io/kubernetes/pkg/conversion/queryparams"
-	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util"
 	"k8s.io/kubernetes/pkg/util/httpstream"
 	"k8s.io/kubernetes/pkg/util/httpstream/spdy"
 )
 
-type upgrader interface {
-	upgrade(*client.Request, *client.Config) (httpstream.Connection, error)
+type Executor interface {
+	Stream(stdin io.Reader, stdout, stderr io.Writer, tty bool) error
 }
 
-type defaultUpgrader struct{}
-
-func (u *defaultUpgrader) upgrade(req *client.Request, config *client.Config) (httpstream.Connection, error) {
-	return req.Upgrade(config, spdy.NewRoundTripper)
+type StreamExecutor interface {
```

Comment:
I was trying to keep the refactor minimal to move things in the right
direction.  That's a big chunk more work on top - I'd prefer to start the
cleanup and move it along, vs trying to change everything with exec.

On Wed, Oct 7, 2015 at 3:40 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/client/unversioned/remotecommand/remotecommand.go
> <https://github.com/kubernetes/kubernetes/pull/14618#discussion_r41435886>
> :
>
> >  }
> >
> > -type defaultUpgrader struct{}
> > -
> > -func (u *defaultUpgrader) upgrade(req *client.Request, config *client.Config) (httpstream.Connection, error) {
> > -	return req.Upgrade(config, spdy.NewRoundTripper)
> > +type StreamExecutor interface {
>
> How would you feel about doing something like
> pkg/client/unversioned/streaming and have a Dialer that returns
> httpstream.Connection, and a either an Executor or Streamer that takes in
> an httpstream.Connection and lets you Stream() stdin/stdout/stderr?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14618/files#r41435886>.
>